src/test/java/bg/healthcheck/BIYD/services/IllnessServiceTests.java

```diff
@@ -0,0 +1,103 @@
+package bg.healthcheck.BIYD.services;
+
+import bg.healthcheck.BIYD.entities.Illnesses;
+import bg.healthcheck.BIYD.repositories.IllnessesRepository;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.dao.DataIntegrityViolationException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
+public class IllnessServiceTests {
+
+    @Mock
+    private IllnessesRepository illnessesRepository;
+
+    private IllnessService illnessService;
+
+    @Before
+    public void before() {
+        illnessService = new IllnessService(illnessesRepository);
+    }
+
+    @Test
+    public void testFindAllIllnessesBySymptomIds() {
+        List<Illnesses> mockFirstIllnesses = new ArrayList<>();
+        mockFirstIllnesses.add(new Illnesses(1L, "first", "first desc"));
+        mockFirstIllnesses.add(new Illnesses(2L, "second", "second desc"));
+        mockFirstIllnesses.add(new Illnesses(3L, "third", "third desc"));
+        List<Illnesses> mockSecondIllnesses = new ArrayList<>();
+        mockFirstIllnesses.add(new Illnesses(4L, "fourth", "first desc"));
+        mockFirstIllnesses.add(new Illnesses(5L, "fifth", "second desc"));
+        mockFirstIllnesses.add(new Illnesses(6L, "sixth", "third desc"));
+        List<Illnesses> mockThirdIllnesses = new ArrayList<>();
+        mockFirstIllnesses.add(new Illnesses(4L, "fourth", "first desc"));
+        mockFirstIllnesses.add(new Illnesses(2L, "second", "second desc"));
+        mockFirstIllnesses.add(new Illnesses(7L, "seventh", "third desc"));
+
+        when(illnessesRepository.findIllnessesBySymptomId(1L)).thenReturn(mockFirstIllnesses);
+        when(illnessesRepository.findIllnessesBySymptomId(2L)).thenReturn(mockSecondIllnesses);
+        when(illnessesRepository.findIllnessesBySymptomId(3L)).thenReturn(mockThirdIllnesses);
```

Comment:
You have misunderstood the concept of the function. findIllnessesBySymptomIds is a function, located in illnessService that receives a List<Integer> symptomIds. From the list of ids, the illnessService.findIllnessesBySymptomIds does what the method says - pulls the Illnesses that contain given symptom ids from the repository. After that, removeDuplicates() removes the duplicate Illnesses from the list. Finally an algorithm goes to sort the final list of Illnesses by most matched symptoms.